apps/opik-backend/src/main/java/com/comet/opik/domain/ProjectMetricsDAO.java

```diff
@@ -116,17 +140,31 @@ public Mono<List<Entry>> getFeedbackScores(@NonNull UUID projectId, @NonNull Pro
                 GET_FEEDBACK_SCORES, "feedbackScores")
                 .flatMapMany(result -> rowToDataPoint(
                         result,
-                        request,
                         row -> row.get("name", String.class),
                         row -> row.get("value", BigDecimal.class)))
                 .collectList());
     }
 
+    @Override
+    public Mono<List<Entry>> getTokenUsage(@NonNull UUID projectId, @NonNull ProjectMetricRequest request) {
+        return template.nonTransaction(connection -> getMetric(projectId, request, connection,
+                GET_TOKEN_USAGE, "token usage")
+                .flatMapMany(result -> rowToDataPoint(
+                        result,
+                        row -> row.get("name", String.class),
+                        row -> row.get("value", Long.class)))
+                .collectList());
+    }
+
     private Mono<? extends Result> getMetric(
             UUID projectId, ProjectMetricRequest request, Connection connection, String query, String segmentName) {
         var template = new ST(query)
-                .add("convert_interval", intervalToSql(request.interval()))
-                .add("is_weekly", request.interval() == TimeInterval.WEEKLY);
+                .add("step", intervalToSql(request.interval()))
+                .add("bucket", wrapWeekly(request.interval(),
+                        "toStartOfInterval(start_time, %s)".formatted(intervalToSql(request.interval()))))
```

Comment:
I would recommend only passing client inputs via bind to avoid SQL injection

Reply:
Agree, however in this case there is no client input involved. All strings are generated in our code.

Reply:
Sorry, I thought because request.interval(), but I see this is not passed directly